### DISCUSSION TITLE
Pixellated Mouse Cursor - Found The Source + Need Fix

### DISCUSSION THREAD

**Discussion by nicolasbulgarides** [AlmasB/FXGL#1045] [Q&A] [answered]
Hello!

I am not the only person to have mentioned this, but I have been eager to replace the default mouse cursor since it is pixellated. After looking through  FXGL core files, I found the location of the file and I also found the code that loads the cursor and on the back end it seems to be way early in the initialization process which makes sense as to why it is hard to change pre-loading so that you dont see a different cursor in loading.

I know that FXGL is open source...but when I tried to edit the class / delete that photo for the cursor and put a different one in there, it told me that the library was read-only. Does anyone know of any workarounds, and if not, Almas, could you add a method somewhere with something like a boolean override + an alternate image source?  Could you clarify if it is possible to edit FXGL classes and if so what does the read only mean in this context?

Thanks everyone

![cursor](https://user-images.githubusercontent.com/50469644/118339523-f1071280-b4cd-11eb-822a-1aafbdb0f723.PNG)

**Comment by AlmasB** [comment]
Hi,

Re: "read-only", I don't know what you mean, perhaps your IDE is telling you that it's a 3rd party library in your project? To contribute to FXGL, please read how to set it up for development, [here](https://github.com/AlmasB/FXGL/blob/dev/CONTRIBUTING.md)

Re: ability to override the default cursor globally. This sounds like a setting that can be added, I will open an issue for this.

Edit: [here's the issue](https://github.com/AlmasB/FXGL/issues/1046). Given this is a simple fix, I've set the milestone to 11.16, which is released on May 25th.

**Comment by AlmasB** [comment, marked as answer]
@nicolasbulgarides This is now implemented, you can use:

```
settings.setDefaultCursor(new CursorInfo("file.png", hotspotX, hotspotY));
```

To get the latest build, you can use this:
```
<dependencies>
    <dependency>
        <groupId>com.github.almasb</groupId>
        <artifactId>fxgl</artifactId>
        <version>dev-SNAPSHOT</version>
    </dependency>
</dependencies>

<repositories>
    <repository>
        <id>oss.sonatype.org-snapshot</id>
        <url>http://oss.sonatype.org/content/repositories/snapshots</url>
    </repository>
</repositories>
```

**Reply by nicolasbulgarides** [reply]
Hi Almas thanks for this - how do I turn that dependency info above from Maven to Gradle...do you have a auto-generated copy for Gradle? Will that method for Cursor Settings be called in the initSettings?

**Reply by AlmasB** [reply]
You can try [this](https://docs.gradle.org/current/userguide/declaring_repositories.html#sec:declaring_custom_repository) to add a custom repo. As for version, just change to `dev-SNAPSHOT`.

Yes, `initSettings()` should do it.
